### PR TITLE
Add support for using lit tests

### PR DESCRIPTION
**Pull request by vchuravy**
Creating the infrastructure for this took me more time
than I would like to admit, but at least this allows us
to add tests that use `lit`. Right now there is only a
simple example. 

Let me know what you think about the structure and naming
of the files.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -22,3 +22,38 @@ end
 haskey(ENV, "CI") && GPUCompiler.timings()
 
 end
+
+using Pkg
+Pkg.add(PackageSpec(name="LLVM_full_jll", version="$(Base.libllvm_version.major)"))
```
Comment (maleadt) [inline test/runtests.jl:27]: :sweat_smile:

File: test/Project.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[deps]
```
Comment (maleadt) [inline test/Project.toml:1]: is this file needed?

**Comment by maleadt** [comment]
```
 FAIL: GPUCompiler :: codegen/native.jl (1 of 1)
******************** TEST 'GPUCompiler :: codegen/native.jl' FAILED ********************
Script:
--
: 'RUN: at line 1';   julia --startup-file=no -L ../definitions/native.jl /builds/JuliaGPU/GPUCompiler.jl/test/codegen/native.jl | FileCheck /builds/JuliaGPU/GPUCompiler.jl/test/codegen/native.jl
--
Exit Code: 2
Command Output (stderr):
--
ERROR: LoadError: ArgumentError: Package LLVM [929cbde3-209d-540e-8aea-75f648917ca0] is required but does not seem to be installed:
 - Run `Pkg.instantiate()` to install all recorded dependencies.
Stacktrace:
 [1] _require(::Base.PkgId) at ./loading.jl:998
 [2] require(::Base.PkgId) at ./loading.jl:927
 [3] require(::Module, ::Symbol) at ./loading.jl:922
 [4] include(::Function, ::Module, ::String) at ./Base.jl:382
 [5] include(::Module, ::String) at ./Base.jl:370
 [6] top-level scope at none:2
 [7] eval at ./boot.jl:331 [inlined]
 [8] eval(::Expr) at ./client.jl:467
 [9] top-level scope at ./none:3
in expression starting at /builds/JuliaGPU/GPUCompiler.jl/src/GPUCompiler.jl:3
ERROR: LoadError: Failed to precompile GPUCompiler [61eb1bfa-7361-4325-ad38-22787b887f55] to /root/.julia/compiled/v1.5/GPUCompiler/yPwef_mwq7P.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1289
 [3] _require(::Base.PkgId) at ./loading.jl:1029
 [4] require(::Base.PkgId) at ./loading.jl:927
 [5] require(::Module, ::Symbol) at ./loading.jl:922
 [6] include(::Function, ::Module, ::String) at ./Base.jl:382
 [7] include at ./Base.jl:370 [inlined]
 [8] exec_options(::Base.JLOptions) at ./client.jl:279
 [9] _start() at ./client.jl:506
in expression starting at /builds/JuliaGPU/GPUCompiler.jl/test/definitions/native.jl:1
```

File: test/Project.toml (hunk 2 above)
Comment (vchuravy) [reply test/Project.toml:1]: Yeah it's the new post-1.3 way of setting up test dependency.

File: test/runtests.jl (hunk 1 above)
Comment (vchuravy) [reply test/runtests.jl:27]: Yeah I need to check whether LLVM_jll has lit and the rest of the tools.
